internal/services/containerapps/helpers/container_apps.go

```diff
@@ -316,9 +316,11 @@ type CustomDomain struct {
 
 func ContainerAppIngressCustomDomainSchema() *pluginsdk.Schema {
 	return &pluginsdk.Schema{
-		Type:     pluginsdk.TypeList,
-		Optional: true,
-		MaxItems: 1,
+		Type:       pluginsdk.TypeList,
+		Optional:   true,
+		Computed:   true,
+		MaxItems:   1,
+		Deprecated: "This property is deprecated in favour of the new `azurerm_container_app_custom_domain` resource and will become computed only in a future release.",
```

Comment:
Should we leave a // TODO 4.0 here for removing `Optional` before the next major release?